### PR TITLE
SDIT-2337: 👽️ React to api changes in detail endpoint

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: integration_tests/mockApis/prisonerDetail.json [hunk 1]
```diff
@@ -18,12 +18,71 @@
...
+    }
+  ],
   "aliases": [
-    { "last": "ALIASA", "first": "OTHERA", "middle": "A", "birthDate": "19800101" },
-    { "last": "ALIASB", "first": "OTHERB", "middle": "B", "birthDate": "19800102" }
+    {
+      "last": "ALIASA",
+      "first": "OTHERA",
+      "middle": "A",
+      "birthDate": "1980-01-01"
```
Comment (petergphillips) [inline integration_tests/mockApis/prisonerDetail.json:52]: note the new date of birth format

File: integration_tests/mockApis/prisonerDetail.json [hunk 2]
```diff
@@ -151,5 +237,24 @@
       "HDCAD": "2001-01-01",
       "HDCED": "2004-01-01"
     }
-  ]
+  ],
+  "sentenceSummary": {
```
Comment (petergphillips) [inline integration_tests/mockApis/prisonerDetail.json:241]: new section

File: server/utils/dateHelpers.test.ts [hunk 3]
```diff
@@ -14,19 +14,6 @@ describe('formatDate', () => {
   })
 })
 
-describe('formatDateNoHyphens', () => {
```
Comment (petergphillips) [inline server/utils/dateHelpers.test.ts:17]: don't need this as doing it now in back end

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**